packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -472,8 +472,17 @@ class _WindowsUtils extends OperatingSystemUtils {
     required super.processManager,
   }) : super._private();
 
+  HostPlatform? _hostPlatform;
+
   @override
-  HostPlatform hostPlatform = HostPlatform.windows_x64;
+  HostPlatform get hostPlatform {
+    if (_hostPlatform == null) {
+      var arch = _platform.environment['PROCESSOR_ARCHITECTURE'] ?? '';
+      _hostPlatform = (arch == 'ARM64') ? HostPlatform.windows_arm64 :
+                                          HostPlatform.windows_x64;
+    }
+    return _hostPlatform!;
+  }
```

Comment:
Can this be unit tested? Perhaps here: https://github.com/flutter/flutter/blob/9f6090c9adb6265feaee68661dcf22fe3a2c5937/packages/flutter_tools/test/general.shard/base/os_test.dart#L146